src/util/request-manager.js

```diff
@@ -287,6 +287,11 @@ export default class RequestManager {
 
         successHosts[parts.hostname] = true;
 
+        if (body && typeof body.error === 'string') {
```

Comment:
This is possibly problematic. In some cases the response is a 40X status with a body that contains an error property. An example is a 404 that has a json body containing error with 'Not Found' (if I recall correctly the registry sometimes returns this).

In the lines below you can see:
```
if (res.statusCode === 400 || res.statusCode === 404) {
  body = false;
}
```
The behavior here is that body is set to false on these status codes, but if the response body happened to contain an error property, the behavior is changed.